lnwallet/btcwallet/btcwallet.go

```diff
@@ -923,18 +923,37 @@ func minedTransactionsToDetails(
 			return nil, err
 		}
 
+		// isOurAddress is a map containing the output indices
+		// controlled by the wallet.
+		// Note: We make use of the information in `MyOutputs` provided
+		// by the `wallet.TransactionSummary` structure that holds
+		// information only if the output is controlled by the wallet.
+		isOurAddress := make(map[int]bool, len(tx.MyOutputs))
+		for _, o := range tx.MyOutputs {
+			isOurAddress[int(o.Index)] = true
+		}
+
 		var destAddresses []btcutil.Address
-		for _, txOut := range wireTx.TxOut {
-			_, outAddresses, _, err := txscript.ExtractPkScriptAddrs(
+		var outputDetails []lnwallet.OutputDetail
+		for i, txOut := range wireTx.TxOut {
+			var addresses []btcutil.Address
+			sc, outAddresses, _, err := txscript.ExtractPkScriptAddrs(
 				txOut.PkScript, chainParams,
 			)
-			if err != nil {
-				// Skip any unsupported addresses to prevent
-				// other transactions from not being returned.
-				continue
+			if err == nil {
+				// Add supported addresses.
+				destAddresses = append(destAddresses, outAddresses...)
+				addresses = outAddresses
 			}
 
-			destAddresses = append(destAddresses, outAddresses...)
+			outputDetails = append(outputDetails, lnwallet.OutputDetail{
+				OutputType:   sc,
+				Addresses:    addresses,
+				PkScript:     txOut.PkScript,
+				OutputIndex:  i,
+				Value:        btcutil.Amount(txOut.Value),
+				IsOurAddress: isOurAddress[i],
+			})
```

Comment:
not sure of this. If `err != nil` then we probably dont want to include the OutputDetail either, right? also, if err != nil then for the above, the output detail added will have a nil output type and nil addresses slice. So i guess my main question is, if we do not support an address, do we still want to create an output detail entry for it? and if so, should we add an "Unsupported" field to the Output detail or something?"

Reply:
ah ok, i see that the `ExtractPkScriptAddrs` func will return a `NonStandard` script class if an error is returned. So this should be g2g 👍